src/Parameter.h

```diff
@@ -0,0 +1,379 @@
+#pragma once
+
+#include <functional>
+#include <utility>
+
+enum EditableKind {
+    kInteger,
+    kUInt8,
+    kDouble,
+    kDoubletime,
+    kCString,
+    kFloat,
+    kEnum
+};
+
+class Parameter {
+    public:
+        Parameter(
```

Comment:
I have not followed closely in the past months, but what has become of the idea to us a `class Parameter : public std::variant<std::int32_t, double, String, ...>`? I think it would implement some of the features already that this class provides